File: bench/bench_iter.py
import time
NA, PA = 100_000, 20
NM, PM = 10_000, 40

arr = [i for i in range(1, NA + 1)]
m = {f"k{i}": i for i in range(1, NM + 1)}

acc = 0
t0 = time.process_time()
for _ in range(PA):
    for v in arr:
        acc += v
for _ in range(PM):
    for k, v in m.items():
        acc += v
t1 = time.process_time()
print(f"python iter {NA*PA + NM*PM} = {acc}  time: {t1-t0:.4f}s")

File: bench/bench_iter.lua
local NA, PA = 100000, 20
local NM, PM = 10000, 40

local arr = {}
for i = 1, NA do arr[i] = i end
local m = {}
for i = 1, NM do m["k" .. i] = i end

local acc = 0
local t0 = os.clock()
for _ = 1, PA do
    for _, v in ipairs(arr) do acc = acc + v end
end
for _ = 1, PM do
    for _, v in pairs(m) do acc = acc + v end
end
local t1 = os.clock()
print(string.format("lua    iter %d = %d  time: %.4fs", NA * PA + NM * PM, acc, t1 - t0))
